### PR TITLE
feat(roll): roll Playwright to v1.44.0

### PR DESCRIPTION
**Pull request by mxschmitt**
https://github.com/microsoft/playwright-python/issues/2385

### CODE REVIEW

File: scripts/expected_api_mismatch.txt [hunk 1]
```diff
@@ -12,3 +12,5 @@ Parameter type mismatch in BrowserContext.route(handler=): documented as Callabl
 Parameter type mismatch in BrowserContext.unroute(handler=): documented as Union[Callable[[Route, Request], Union[Any, Any]], None], code has Union[Callable[[Route, Request], Any], Callable[[Route], Any], None]
 Parameter type mismatch in Page.route(handler=): documented as Callable[[Route, Request], Union[Any, Any]], code has Union[Callable[[Route, Request], Any], Callable[[Route], Any]]
 Parameter type mismatch in Page.unroute(handler=): documented as Union[Callable[[Route, Request], Union[Any, Any]], None], code has Union[Callable[[Route, Request], Any], Callable[[Route], Any], None]
+
+Parameter type mismatch in Page.add_locator_handler(handler=): documented as Callable[[Locator], Any], code has Callable
```
Comment (mxschmitt) [inline scripts/expected_api_mismatch.txt:16]: I think if we change that, its considered a breaking change (pyright), not sure if we can do it without breaking:

```text
  /Users/maxschmitt/Developer/playwright-python/tests/sync/test_page_add_locator_handler.py:189:66 - error: Argument of type "() -> None" cannot be assigned to parameter "handler" of type "(Locator) -> Any" in function "add_locator_handler"
    Type "() -> None" cannot be assigned to type "(Locator) -> Any"
      Function accepts too many positional parameters; expected 0 but received 1 (reportGeneralTypeIssues)
```

File: playwright/_impl/_page.py [hunk 2]
```diff
@@ -1270,48 +1289,72 @@ async def set_checked(
...
-                    await coro_or_future
-
+            handler = self._locator_handlers.get(uid)
+            if handler and handler.times != 0:
+                if handler.times is not None:
+                    handler.times -= 1
+                if self._dispatcher_fiber:
+                    handler_finished_future = self._loop.create_future()
+
+                    def _handler() -> None:
```
Comment (yury-s) [inline playwright/_impl/_page.py:1324]:
```suggestion
                    def handler() -> None:
```

File: playwright/_impl/_page.py [hunk 3]
```diff
@@ -1270,48 +1289,72 @@ async def set_checked(
...
-                        "resolveLocatorHandlerNoReply", {"uid": uid}
+                        "resolveLocatorHandlerNoReply", {"uid": uid, "remove": remove}
                     ),
                     is_internal=True,
                 )
             except Error:
                 pass
 
+    async def remove_locator_handler(self, locator: "Locator") -> None:
+        for uid, data in self._locator_handlers.copy().items():
```
Comment (yury-s) [inline playwright/_impl/_page.py:1353]: does python not support deletion during iteration?

**Review by yury-s — APPROVED** [review]

File: playwright/_impl/_page.py [hunk 4]
```diff
@@ -98,6 +98,25 @@
...
+
+    def __init__(
+        self, locator: "Locator", handler: Callable[..., Any], times: Union[int, None]
+    ) -> None:
+        self.locator = locator
+        self._handler = handler
+        self.times = times
+
+    def __call__(self) -> Any:
+        arg_count = len(inspect.signature(self._handler).parameters)
```
[the remark this answers is not in the document]
Comment (mxschmitt) [reply playwright/_impl/_page.py:114]: We treat it the same as for route() handling. In route() its either route or route + request.

**Pull request merged**